dipy/tracking/utils.py

```diff
@@ -595,6 +596,45 @@ def target(streamlines, target_mask, affine, include=True):
             yield sl
 
 
+@_with_initialize
+def target_line_based(streamlines, target_mask, affine=None, include=True):
```

Comment:
Might be useful in the doc to refer to Bresenham's algorithm and my paper 

```
Houde et al. How to avoid biased streamlines-based metrics for streamlines with variable step sizes, ISMRM 2015.
```